### ISSUE TITLE
Use vim-mode keybindings in finder

### ISSUE THREAD

**Issue by muhmud** [junegunn/fzf#3015]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf (0.34.0-1)
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish

## Problem / Steps to reproduce

I'm trying to customize the key bindings used in the finder. What I would like is for them to work in the same [zsh-vim-mode](https://github.com/softmoth/zsh-vim-mode). It would be great if I could start in normal mode, when the finder first comes up.

I can't find anyway to do this, either from the man page or online.

**Comment by junegunn** [comment]
fzf doesn't support any modes. fzf is usually a very short-lived process (doesn't last a few keystrokes during a few seconds), and I believe mode switching is overkill for its purpose.

**Comment by muhmud** [comment]
I know what you mean, that it's usually short-lived, however, it's for those occasions when it's not that I really wanted this.

For example, I tend to use fzf in combination with rg, with the enter key bound to open the file in neovim using neovim-remote. In this case, fzf stays open for a quite a while, and it's tiresome to scroll around by additionally having to hold down the Ctrl key. I prefer this to using it directly from within vim.

**Comment by junegunn** [comment]
I see. Adding a mode is a significant amount of work and I personally don't need it so it won't happen in a foreseeable future (at least by me).

**Comment by muhmud** [comment]
I've completed an implementation of this, and it seems to work really well.

Would you mind if I opened a PR?

**Comment by junegunn** [comment]
Sorry, I still think it's an overkill. And I don't want to make the program any harder to comprehend by adding another mode that implements a small subset of Vim bindings. Actually, I believe Emacs readline bindings, most of which are already implemented, are better suited for fzf (for the same reason, I don't use Vim mode in my shell). And if you really don't want to touch arrow keys, then you can use CTRL-J and CTRL-K and I think that's good enough.

**Comment by junegunn** [comment]
FWIW, here is a poor man's browser mode :)
```sh
fzf --bind j:down,k:up --bind 'start:unbind(j)+unbind(k)' \
    --bind 'esc:rebind(j)+rebind(k)+change-prompt(Browse mode> )' \
    --bind 'i:unbind(j)+unbind(k)+change-prompt(> )'
```

**Comment by muhmud** [comment]
Oh, that's cool that you can do that!

The main reason for having this for me was so that I could browse the results in a similar to way to using `less`. I never could get used to the emacs key bindings, and use vim mode in my shell instead. Also, most of the hard work for this had already been done, i.e. there were already actions that I could bind to for everything.

I was hoping that I wouldn't have to maintain a separate code base, but it might not be so bad; I can now add a few extra quality-of-life hacks I've been wanting...

**Comment by niksingh710** [comment]
@muhmud does your implementation enables editing too in the same mode with a few vim keys?

**Comment by muhmud** [comment]
Yes, by default there are some vim keys to move around the query string, however, I made these capitalized, so `W` instead of `w`. This way none of them interfere with the keys to browse around the results, which I wanted to work like `less`. There are also keys like `I`, `i`, `a`, `0`, `$`, etc.

You can override all of these though, as I also implemented a `--bind-browse` option.
